core/eolearn/tests/test_core_tasks.py

```diff
@@ -229,49 +230,48 @@ def test_duplicate_feature_fails(patch: EOPatch) -> None:
         DuplicateFeatureTask((FeatureType.DATA, "bands", "bands"))(patch)
 
 
-def test_initialize_feature(patch):
-    patch = DeepCopyTask()(patch)
+@pytest.mark.parametrize(
+    "init_val, shape, feature_spec",
+    [
+        (8, (5, 2, 6, 3), (FeatureType.MASK, "test")),
+        (9, (1, 4, 3), (FeatureType.MASK_TIMELESS, "test")),
+        (7, (5, 2, 7, 4), {FeatureType.MASK: ["F1", "F2", "F3"]}),
+    ],
+)
+def test_initialize_feature(
+    init_val: float,
+    shape: Union[Tuple[int, ...], FeatureSpec],
+    feature_spec: FeaturesSpecification,
+    patch: EOPatch,
+) -> None:
+    expected_data = init_val * np.ones(shape)
+    patch = InitializeFeatureTask(feature_spec, shape=shape, init_value=init_val)(patch)
 
-    init_val = 123
-    shape = (5, 10, 10, 3)
-    compare_data = np.ones(shape) * init_val
+    assert all([np.array_equal(patch[f_type][names], expected_data) for f_type, names in parse_features(feature_spec)])
```

Comment:
you can probably do `patch[feature] for feature in parse_features(feature_spec)`

Reply:
yes, done